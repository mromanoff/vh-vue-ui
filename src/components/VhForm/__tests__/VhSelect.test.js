import { shallowMount } from '@vue/test-utils';
import VhSelect from '../VhSelect.vue';

describe('VhSelect.vue', () => {
  let wrapper;

  it('renders default VhSelect', () => {
    wrapper = shallowMount(VhSelect, {
      propsData: {
        name: 'Test Name',
        label: 'Test Label',
        options: [],
      },
    });
    expect(wrapper.props().label).toBe('Test Label');
    expect(wrapper.props().labelHidden).toBe(false);
    expect(wrapper.props().name).toBe('Test Name');
    expect(wrapper.props().value).toBe('');
    expect(wrapper.props().options).toEqual([]);
    expect(wrapper.props().disabled).toBe(false);
    expect(wrapper.props().state).toBe('pristine');
    expect(wrapper.props().message).toBe('');
    expect(wrapper.props().size).toBe('medium');
    expect(wrapper.props().theme).toBe('light');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhSelect', () => {
    wrapper = shallowMount(VhSelect, {
      propsData: {
        name: 'Test Name',
        label: 'Test Label',
        labelHidden: true,
        value: 'any',
        disabled: true,
        state: 'dirty',
        size: 'large',
        theme: 'dark',
        message: 'This is a test',
        options: ['stateItem'],
      },
    });
    expect(wrapper.props().label).toBe('Test Label');
    expect(wrapper.props().labelHidden).toBe(true);
    expect(wrapper.props().name).toBe('Test Name');
    expect(wrapper.props().value).toBe('any');
    expect(wrapper.props().options).toEqual(['stateItem']);
    expect(wrapper.props().disabled).toBe(true);
    expect(wrapper.props().state).toBe('dirty');
    expect(wrapper.props().message).toBe('This is a test');
    expect(wrapper.props().size).toBe('large');
    expect(wrapper.props().theme).toBe('dark');
    expect(wrapper).toMatchSnapshot();
  });
});
