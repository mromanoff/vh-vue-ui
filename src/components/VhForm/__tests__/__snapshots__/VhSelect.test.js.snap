// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhSelect.vue renders default VhSelect 1`] = `
<div class="Form-field Form-field--medium Form-field--light Form-field--pristine"><label for="Test Name" class="Form-label">
    Test Label
  </label> <select id="Test Name" class="Form-input"></select>
  <p class="Form-message" style="display: none;">

  </p>
</div>
`;

exports[`VhSelect.vue renders modified VhSelect 1`] = `
<div class="Form-field Form-field--large Form-field--dark Form-field--dirty"><label for="Test Name" class="Form-label u-screenReaderOnly">
    Test Label
  </label> <select id="Test Name" disabled="disabled" class="Form-input">
    <option value="">

    </option>
  </select>
  <p class="Form-message">
    This is a test
  </p>
</div>
`;
